Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -823,7 +816,7 @@ def audit_specs
       stable_url_minor_version = stable_url_version.minor.to_i
 
       formula_suffix = stable.version.patch.to_i
-      throttled_rate = THROTTLED_FORMULAE[formula.name]
+      throttled_rate = audit_exception_list("THROTTLED_FORMULAE")[formula.name]
```

Comment:
```suggestion
      throttled_rate = tap_audit_exception("THROTTLED_FORMULAE")[formula.name]
```

but even then that doesn't seem quite right to be returning an integer. Hmm.

Reply:
I renamed the method `tap_audit_exception_list`. The method takes in a list name and returns the entire list. This is two few use cases:

1. In this case, we need to get the value of the hash so we get the list using this method and then access the `formula.name` element:
    ```ruby
    tap_audit_exception_list(:throttled_formulae)[formula.name]
    ```
1. In a case where it's simply an allow/blocklist, we would use call `.include?` on the list returned from this method:
    ```ruby
    tap_audit_exception_list(:versioned_keg_only_allowlist).include? formula.name
    ```

Reply:
Makes sense to me. I wonder if `tap_audit_exception?(:throttled_formulae, formula.name)` would be a nicer API for handling the case where the file and/or key don't exist as expected?

Reply:
Here's a table with the possibilities for having this in one method:

||Exception List is an Array|Exception List is a Hash|
|---|:-:|:-:|
|Formula is in the list|`true`|the value|
|Formula is **not** in the list|`false`|`false`|

Two usage examples:

```ruby
problem if tap_audit_exception(:versioned_keg_only_allowlist, formula.name)
```

```ruby
throttle_rate = tap_audit_exception(:throttle_rate, formula.name)
problem if throttle_rate && formula.version % throttle_rate != 0
```

Is that what you're thinking?

Reply:
> Here's a table with the possibilities for having this in one method:

Thanks!

> ```ruby
> problem if tap_audit_exception(:versioned_keg_only_allowlist, formula.name)
> ```

This looks good 👍🏻 


> ```ruby
> throttle_rate = tap_audit_exception(:throttle_rate, formula.name)
> problem if throttle_rate && formula.version % throttle_rate != 0
> ```

This looks good too. If there's multiple usage of the same pattern here it could be a more specific `tap_audit_throttle(:throttle_rate, formula.name, formula.version)` or similar (but that doesn't seem to be the case).

Reply:
Yeah, I like that. I'll need to look at the existing lists and see if this would be useful.

Reply:
Okay, with the method being `tap_audit_throttle(list, formula, value)`, this would be the new table:

||`formula` in list|`formula` not in list|
|---|:-:|:-:|
|`value` is `nil`|`true` or value (if hash)|`false`|
|`value` matches|`true`|`false`|
|`value` doesn't match|`false`|`false`|

Not passing a `value` to the method will return the value if it's a hash and return `true` if it's an array

Reply:
Sounds good!